config/initializers/active_record_extensions.rb

```diff
@@ -53,6 +52,43 @@ def destroy_annotations_and_versions
     self.annotations.destroy_all if self.respond_to?(:annotations)
   end
 
+  def sent_to_slack
+    @sent_to_slack
+  end
+
+  def sent_to_slack=(bool)
+    @sent_to_slack = bool
+  end
+
+  def to_slack(text)
+    # https://api.slack.com/docs/message-formatting#how_to_escape_characters
+    { '&' => '&amp;', '<' => '&lt;', '>' => '&gt;' }.each { |k,v|
+      text = text.gsub(k,v)
+    }
+    text
+  end
+
+  def to_slack_url(url, text)
+    url.insert(0, "#{CONFIG['checkdesk_client']}/") unless url.start_with? "#{CONFIG['checkdesk_client']}/"
+    text = self.to_slack(text)
+    text = text.tr("\n", ' ')
+    "<#{url}|#{text}>"
+  end
+
+  def to_slack_quote(text)
+    text = I18n.t(:blank) if text.blank?
+    text = self.to_slack(text)
+    text.insert(0, "\n") unless text.start_with? "\n"
+    text.gsub("\n", "\n>")
+  end
```

Comment:
Just a suggestion: I think that methods from line 63 until line 83 could be converted to methods of the Slack bot. They don't use any information of the `ActiveRecord` model that inherits them.